mmv1/third_party/terraform/website/docs/functions/project_from_id.html.markdown

```diff
@@ -0,0 +1,42 @@
+---
+page_title: project_from_id Function - terraform-provider-google
+description: |-
+  Returns the project within a provided resource id, self link, or OP style resource name.
+---
+
+# Function: project_from_id
+
+Returns the project within a provided resource id, self link, or OP style resource name.
+
+For more information about using provider-defined functions with Terraform [see the official documentation](https://developer.hashicorp.com/terraform/plugin/framework/functions/concepts).
+
+## Example Usage
+
+```terraform
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+    }
+  }
+}
+
+# Value is "my-project"
+output "function_output" {
+  value = provider::google::project_from_id("https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1-c/instances/my-instance")
+}
+```
+
+## Signature
+
+```text
+project_from_id(id string) string
+```
+
+## Arguments
+
+1. `id` (String) A string of a resource's id, a resource's self link, or an OP style resource name. For example, these are all valid values:
```

Comment:
I don't know what the correct term to use here is w.r.t "OP style resource name". I got that phrase from here (https://github.com/hashicorp/terraform-provider-google/issues/16244#issuecomment-1867033535) and I've found [docs for Cloud Asset Inventory that list this type of resource name](https://cloud.google.com/asset-inventory/docs/resource-name-format) but I'm not really satisfied. I'm not sure what OP stands for above!

Reply:
https://en.wikipedia.org/wiki/URL#prurl is closest I've got

Reply:
Thank you for this - I've updated the documentation using that new phrasing!